### ISSUE TITLE
[Enhancement] EnumToIntConverter

### ISSUE THREAD

**Issue by Hottemax** [xamarin/XamarinCommunityToolkit#1197]
## Summary

Such a converter would be useful when binding an enum property and a collection of valid enum values to a control with a SelectedIndex, such as a Picker control

## API

Add new class `EnumToIntConverter : IValueConverter` in the _Xamarin.CommunityToolkit.Converters_ namespace
Conversion would work along the lines of
```csharp
public object Convert(object value, Type targetType, object parameter, CultureInfo culture) => value is Enum ? (int)value : 0;
public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture) => value is int enumIntVal && Enum.IsDefined(targetType, enumIntVal) ? Enum.ToObject(targetType, enumIntVal) : 0;
```

## Intended Use Case
`<Picker ItemsSource="{Binding CollectionOfValidEnumsForPicker}" SelectedIndex="{Binding SelectedEnum, Converter={StaticResource EnumToIntConverter}}" />`

## Who Will Do The Work?

- [x] I am willing to take this on myself
- [ ] Just putting this out there for someone to pick up

**Comment by pictos** [comment]
@Hottemax can you update your last comment with a spec?

**Comment by Hottemax** [comment]
> @Hottemax can you update your last comment with a spec?

Sorry, my first PR here in XCR, can you clarify what you mean by "update with a spec")?

**Comment by pictos** [comment]
@Hottemax I want to see how you're thinking about the API

something like this
https://github.com/xamarin/XamarinCommunityToolkit/issues/556

**Comment by Hottemax** [comment]
> @Hottemax I want to see how you're thinking about the API
> 
> something like this
> #556

I have updated the description to provide a little more detail, not quite sure what more to specify.

**Comment by Hottemax** [comment]
Not sure how to "link" the pull request to the issue (apart from mentioning it in the description) - it does not show up under "linked pull requests"

**Comment by jfversluis** [comment]
This is implemented in #1203
